tests/integration/integration-tests.yml

```diff
@@ -4079,6 +4079,95 @@
           - test_five['circuit_termination']['pp_info'] == "PP10-24"
           - test_five['circuit_termination']['description'] == "Test description"
           - test_five['msg'] == "circuit_termination test_circuit_a deleted"
+          -
+
+##
+##
+### NETBOX_SERVICE
+##
+##
+
+    - name: "1 - Device with required information needs to add new service"
+      netbox_device:
+        netbox_url: "http://localhost:32768"
+        netbox_token: "0123456789abcdef0123456789abcdef01234567"
+        data:
+          name: "FOR_SERVICE"
+          device_type: "Cisco Test"
+          device_role: "Core Switch"
+          site: "Test Site"
+          status: "Staged"
+        state: present
+
+    - name: "NETBOX_SERVICE: Create new service"
+      netbox_service:
+        netbox_url: "http://localhost:32768"
+        netbox_token: "0123456789abcdef0123456789abcdef01234567"
+        data:
+          device: "FOR_SERVICE"
+          name: "node-exporter"
+          port: 9100
+          protocol: TCP
+        state: present
+      register: test_service
+
+    - name: "NETBOX_SERVICE ASSERT - Create"
+      assert:
+        that:
+          - test_service is changed
+
+    - name: "NETBOX_SERVICE: Test idempotence"
+      netbox_service:
+        netbox_url: "http://localhost:32768"
+        netbox_token: "0123456789abcdef0123456789abcdef01234567"
+        data:
+          device: "FOR_SERVICE"
+          name: "node-exporter"
+          port: 9100
+          protocol: TCP
+        state: present
+      register: test_service_idempotence
+
+    - name: "NETBOX_SERVICE ASSERT - Not changed"
+      assert:
+        that:
+          - test_service_idempotence['msg'] == "services node-exporter already exists"
+
+    - name: "NETBOX_SERVICE: Test update"
+      netbox_service:
+        netbox_url: "http://localhost:32768"
+        netbox_token: "0123456789abcdef0123456789abcdef01234567"
+        data:
+          device: "FOR_SERVICE"
+          name: "node-exporter"
+          port: 9100
+          protocol: UDP
+        state: present
+      register: test_service_update
+
+    - name: "NETBOX_SERVICE ASSERT - Service has been updated"
+      assert:
+        that:
+          - test_service_update is changed
+
+
+    - name: "NETBOX_SERVICE: Test service deletion"
+      netbox_service:
+        netbox_url: "http://localhost:32768"
+        netbox_token: "0123456789abcdef0123456789abcdef01234567"
+        data:
+          device: "FOR_SERVICE"
+          name: "node-exporter"
```

Comment:
Thanks for all the changes! Do you mind adding more to the asserts? If you look at each type of test for the other modules, we're attempting to validate some of the data as well. I would just look at a few of the tests for each create, idempotent, update, delete.

Everything else looks OK.

Reply:
hi, thanks for all your comments.
Test asserts have been updated